build.zig

```diff
@@ -44,6 +45,24 @@ pub fn build(b: *std.Build) void {
     }
     run_step.dependOn(&run_cmd.step);
 
+    // Add snapshot tool
+    const snapshot_exe = b.addExecutable(.{
+        .name = "snapshot",
+        .root_source_file = b.path("src/snapshot.zig"),
+        .target = target,
+        .optimize = optimize,
+        .link_libc = true,
+    });
+    b.installArtifact(snapshot_exe);
```

Comment:
nit: do we want this to build by default? An install step makes this always build with `zig build`. One alternative is to use a flag instead. `zig build -Dsnapshot` and only install when that flag is used. I think we can make the flag automatically get set to true when `zig build snapshot` is called.